python/neuroglancer/pipeline/storage.py

```diff
@@ -90,11 +103,8 @@ def put_file(self, file_path, content, compress=False):
             content = self._compress(content)
 
         if self._n_threads:
-            try:
-                self._queue.put(('put_file', file_path, content, compress))
-            except Queue.Full:
-                self.wait_until_queue_empty()
-                self._queue.put(('put_file', file_path, content, compress))
+            # None is the non-existant callback
+            self._queue.put(('put_file', None, file_path, content, compress), block=True)
```

Comment:
I could go either way on this one (and I have in the past in various JS codebases). Whichever one you like better. :)